gax/src/main/java/com/google/api/gax/rpc/ApiExceptions.java

```diff
@@ -43,15 +42,22 @@ private ApiExceptions() {}
    * will be {@link UncheckedExecutionException}), the exception is processed in the following way:
    *
    * <ol>
-   *   <li>If the exception cause is a RuntimeException, the RuntimeException is rethrown.
+   *   <li>If the exception cause is a RuntimeException, the RuntimeException is rethrown. To ease
+   *       debugging, the current stacktrace will be preserved as a suppressed exception.
    *   <li>Otherwise, the UncheckedExecutionException is rethrown.
    * </ol>
    */
   public static <ResponseT> ResponseT callAndTranslateApiException(ApiFuture<ResponseT> future) {
     try {
       return Futures.getUnchecked(future);
     } catch (UncheckedExecutionException exception) {
-      Throwables.throwIfInstanceOf(exception.getCause(), RuntimeException.class);
+      if (exception.getCause() instanceof RuntimeException) {
+        RuntimeException cause = (RuntimeException) exception.getCause();
+        // To ensure that the caller stacktrace is not lost, add a suppressed exception here.
+        cause.addSuppressed(new RuntimeException("Asynchronous task failed"));
```

Comment:
Since it is repeated in at least 2 places and needs string to explain what actually happened here (`"Asynchronous task failed"`) it probably has to be an actual new unchecked exception type (also makes it easier to handle programmatically the suppressed exception case).